Added colored indication for project versions compatibility inside Project Manager

syskrank opened the pull request:
Addressed the [#25497](https://github.com/godotengine/godot/issues/25497) by adding a simple indication on how the projects are compatible with current instance of Godot Engine basing on **config_version** parameter for **project.godot**.

qarmin commented:
I have some ideas about it:
1. Deleted project have a label "Older version project", but if I remember this correctly, projects were created in 3.1 version.
2. When project is selected, text overlap with right edge of selection
3. The text "Compatible" and "Incompatible" isn't clear. I would add version of engine to each of project e.g. "Incopatible. Created in Godot 4.0" or something similar

![bez tytulu](https://user-images.githubusercontent.com/41945903/52042867-8de05480-253e-11e9-8c2d-c512998eda80.png)

groud commented:
I think the "Compatible" label is not necessary. I would only keep the label for "incompatible" projects.

syskrank commented:
I've updated the overall logic of this thing and modified the text alignment.
Compatible projects are now just silent, however issues are highlighted.
Here it is right now:

![image](https://user-images.githubusercontent.com/11526664/52069628-2a7f1280-2590-11e9-88d5-3d55acbfae40.png)

akien-mga commented on editor/project_manager.cpp:
```diff
@@ -1400,6 +1401,40 @@ void ProjectManager::_load_recent_projects() {
 		fpath->set_clip_text(true);
 
+		// indicate project compatibility via colored label
+		Label *engine_version = memnew(Label());
```
Should be `memnew(Label)`. Setting the node name is probably not necessary.

akien-mga replied:
> Setting the node name is probably not necessary.

Well it's done for other controls above, so I guess it's fine here :)

akien-mga commented on editor/project_manager.cpp:
```diff
@@ -1400,6 +1401,40 @@ void ProjectManager::_load_recent_projects() {
+		}
+
+		// show warning if something is odd
+		if (indicate_incompat) {
```
It's quite wasteful to allocate `engine_version` for every project, and discard it if not needed. Apart from short transitions periods between releases where the `config_version` changed (which should not happen often, it's not linked to the engine version), most users won't have incompatible projects.

It would be cleaner and more efficient to check first if a label is needed, and instantiate it only then.

merumelu commented on editor/project_manager.cpp:
```diff
@@ -1400,6 +1401,41 @@ void ProjectManager::_load_recent_projects() {
+			Label *engine_version = memnew(Label);
+			engine_version->set_name("engine_version");
+			engine_version->set_h_size_flags(SIZE_EXPAND_FILL);
+			Color engine_color = Color(1, 0, 0, 0.9);
```
This should use `get_color("error_color", "Editor")`.

merumelu commented on editor/project_manager.cpp:
```diff
@@ -1400,6 +1401,41 @@ void ProjectManager::_load_recent_projects() {
+			// detailed check
+			if (FileAccess::exists(item.conf)) {
+				if (item.config_version < ProjectSettings::CONFIG_VERSION) {
+					engine_color = Color(1, 1, 0.2, 0.5);
```
And here `get_color("warning_color", "Editor")`.
